tests/unit/types/arrays/test_neuraloperations.py

```diff
@@ -0,0 +1,177 @@
+import pytest
+import copy
+
+import numpy as np
+
+from jina.types.arrays.neuraloperations import (
+    cosine_distance,
+    euclidean_distance_squared,
+)
+from jina.types.arrays.memmap import DocumentArrayMemmap
+from jina import Document, DocumentArray
+
+
+@pytest.fixture
+def docarrays_for_embedding_distance_computation():
+    d1 = Document(embedding=np.array([0, 0, 0]))
+    d2 = Document(embedding=np.array([3, 0, 0]))
+    d3 = Document(embedding=np.array([1, 0, 0]))
+    d4 = Document(embedding=np.array([2, 0, 0]))
+
+    d1_m = Document(embedding=np.array([1, 0, 0]))
+    d2_m = Document(embedding=np.array([2, 0, 0]))
+    d3_m = Document(embedding=np.array([0, 0, 1]))
+    d4_m = Document(embedding=np.array([0, 0, 2]))
+    d5_m = Document(embedding=np.array([0, 0, 3]))
+
+    D1 = DocumentArray([d1, d2, d3, d4])
+    D2 = DocumentArray([d1_m, d2_m, d3_m, d4_m, d5_m])
+    return D1, D2
+
+
+@pytest.fixture
+def numpy_array():
+    return np.array([[1, 0, 0], [2, 0, 0], [3, 0, 0]])
+
+
+@pytest.fixture
+def numpy_array_query():
+    return np.array([[1, 0, 0]])
+
+
+def test_new_distances_equal_previous_distances():
```

Comment:
How about we test this not against our own (previous?) home-cooked functions, but against stable libraries (`scipy` or `scikit-learn`)? Would it be possible to do that, and add them as a test-only dependency (if they are not already a base dpendency)?

Reply:
I already compared with scipy cdist library and the results are the same, from our own previous codebase and the new proposal.

Reply:
Looking at [here](https://github.com/jina-ai/jina/blob/master/extra-requirements.txt) it seems there is already a way to add a test-only dependency (the `cicd` tag), and scipy is in fact already there